src/parser/generateDocuments.ts

```diff
@@ -0,0 +1,68 @@
+import * as ts from "typescript";
+import type { ParserOptions } from "./";
+import { iterateSymbolTable } from "./utilities";
+import { ComponentDoc } from "./types";
+import { Parser } from "./";
+
+export const fileToDocumentList =
+  (parser: Parser, parserOpts: ParserOptions, checker: ts.TypeChecker) =>
+  (acc: ComponentDoc[], sourceFile: ts.SourceFile) => {
+    const moduleSymbol = checker.getSymbolAtLocation(sourceFile);
+    if (!moduleSymbol) {
+      return acc;
+    }
+
+    const components = checker.getExportsOfModule(moduleSymbol);
```

Comment:
Now that you’re here you could take this a step further and flatMap over this list and combine it with the return from `iterateSymbolTable`